examples/vision/attention_mll_classification.py

```diff
@@ -0,0 +1,541 @@
+from tensorflow.python.keras import initializers, regularizers
+from matplotlib import pyplot as plt
+
+os.environ["TF_CPP_MIN_LOG_LEVEL"] = "3"
+plt.style.use("ggplot")
+
+"""
+## Create Dataset
+
+In this example, either of `mnist` or `fashion_mnist` data can be used.
+At first we will set up the configurations, create the function for preparing them and then load the datasets that will be *used*.
+
+### Configurations
+- DATASET: Either mnist digits or fashion mnist data.
+- POS_CLASS: The desired class to be kept in the positive bag.
+- BAG_COUNT: The number of training bags.
+- VAL_BAG_COUNT: The number of validation bags.
+- BAG_SIZE: The number of instances in a bag.
+- PLOT_SIZE: The number of bags to plot.
+- ENSEMBLE_AVG_COUNT: The number of models to create and get their average. (OPTIONAL: often results in better performance - set to 1 for single model)
+"""
+
+DATASET = {"mnist": mnist, "fashion_mnist": fashion_mnist}
+POS_CLASS = 1
+BAG_COUNT = 1000
+VAL_BAG_COUNT = 300
+BAG_SIZE = 3
+PLOT_SIZE = 3
+ENSEMBLE_AVG_COUNT = 5
+
+"""
+### Prepare Bags
+This function returns the bags along with their corresponding labels.
+
+Since the attention operator is a permutation-invariant operator,
+the desired/positive class is randomly placed among the instances
+in the positive bag.
+"""
+
+
+def create_bags(x, y, pos_class, bag_count, instance_count):
+
+    # Set up bags.
+    bags = []
+    bag_labels = []
+
+    # Normalize input data.
+    x = np.divide(x, 255.0)
+
+    # Count positive samples.
+    count = 0
+
+    # Take out the filter for specific class.
+    filter_class = np.where(y == pos_class)[0]
+
+    # Assign new variables consisting of this class.
+    x_pos_class = x[filter_class]
+    y_pos_class = y[filter_class]
+
+    # From overall data, remove this class.
+    x_neg_classes = np.delete(x, filter_class, 0)
+    y_neg_classes = np.delete(y, filter_class, 0)
+
+    # Merge both inputs and labels to each another.
+    x_merged = np.concatenate([x_pos_class, x_neg_classes], axis=0)
+    y_merged = np.concatenate([y_pos_class, y_neg_classes], axis=0)
+
+    # Data are ordered in such a way: [pos_class... neg_classes]
+    # Shuffle the data randomly.
+    order = np.arange(len(x_merged))
+    np.random.shuffle(order)
+    x = x_merged[order]
+    y = y_merged[order]
+
+    for _ in range(bag_count):
+
+        # Pick a fixed size random subset of samples.
+        index = np.random.choice(x.shape[0], instance_count, replace=False)
+        images = x[index]
+        labels = y[index]
+
+        # By default, all bags are labelled as 0.
+        label = 0
+
+        # Check if there is atleast a positive class in the bag.
+        if pos_class in labels:
+
+            # Positive bag will be labelled as 1.
+            label = 1
+
+            # Increment count by 1.
+            count += 1
+
+        bags.append(images)
+        bag_labels.append(np.array([label]))
+
+    print(
+        "Positive bags: " + str(count),
```

Comment:
Use f strings for string formatting rather than `+` and `str`

Reply:
Updated.